### PR TITLE
ocamlPackages.irmin*: 2.1.0 → 2.2.0

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

new major release <https://github.com/mirage/irmin/releases/tag/2.2.0>

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build ocamlPackages.irmin-unix ocamlPackages.irmin-http

File: pkgs/development/ocaml-modules/irmin/default.nix [hunk 1]
```diff
@@ -27,5 +22,4 @@ buildDunePackage rec {
     license = lib.licenses.isc;
     maintainers = [ lib.maintainers.vbgl ];
   };
-
-}
+} // (import ./src.nix { inherit fetchurl; }))
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/irmin/default.nix:25]: Why don’t you inherit the `src` and `version` attributes from `ppx_irmin`? I find this clearer but maybe it is just a matter of taste.

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/irmin/default.nix]: I agree that's cleaner, as we don't have to duplicate the `meta` set. Thanks!

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.irmin-unix

**Pull request merged**